KachnaOnline.App/Controllers/EventsController.cs

```diff
@@ -144,6 +150,164 @@ public async Task<ActionResult<ManagerEventDto>> PlanEvent(BaseEventDto newEvent
+            }
+            catch (NotAnEventsManagerException)
+            {
+                // Shouldn't happen.
+                return this.ForbiddenProblem();
+            }
+            catch (EventReadOnlyException)
+            {
+                return this.ConflictProblem("The specified event has already passed and cannot be updated.");
+            }
+        }
+
+        /// <summary>
+        /// Sets (overrides) the current linked states to the event specified by <paramref name="eventId"/> with <paramref name="plannedStatesToLink"/>.
+        /// </summary>
+        /// <param name="eventId">Event to set linked planned states for.</param>
+        /// <param name="plannedStatesToLink">A list of planned state IDs to link to the event specified by <paramref name="eventId"/>.</param>
+        /// <response code="204">The planned states were linked to the event.</response>
+        /// <response code="404">The event with the given ID <paramref name="eventId"/> does not exist.</response>
+        /// <response code="409">The event from the past cannot be modified.</response>
+        [HttpPut("{eventId}/linked_states")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+        public async Task<ActionResult> SetLinkedPlannedStatesForEvent(int eventId, PlannedStatesToLinkDto plannedStatesToLink)
+        {
+            try
+            {
+                await _eventsFacade.SetLinkedPlannedStatesForEvent(eventId, plannedStatesToLink);
+                return this.NoContent();
+            }
+            catch (EventNotFoundException)
+            {
+                return this.NotFoundProblem("The specified event does not exist.");
+            }
+            catch (NotAnEventsManagerException)
+            {
+                // Shouldn't happen.
+                return this.ForbiddenProblem();
+            }
+            catch (EventReadOnlyException)
+            {
+                return this.ConflictProblem("The specified event has already passed and cannot be updated.");
+            }
+        }
+
+        /// <summary>
+        /// Clears (unlinks) the current linked states from the event specified by <paramref name="eventId"/>.
+        /// </summary>
+        /// <param name="eventId">Event to unlink linked planned states from.</param>
+        /// <response code="204">The planned states were linked to the event.</response>
+        /// <response code="404">The event with the given ID <paramref name="eventId"/> does not exist.</response>
+        /// <response code="409">The event from the past cannot be modified.</response>
+        [HttpDelete("{eventId}/linked_states")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
```

Comment:
Fix these so that they match the actual responses produced by this controller method.

Reply:
Yes, the controller documentation and its method decorators are still work-in-progress a bit. I am working on updating these right now.